packages/realm/src/Logger.ts

```diff
@@ -0,0 +1,91 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { assert, binding } from "./internal";
+
+export type LogLevel = "all" | "trace" | "debug" | "detail" | "info" | "warn" | "error" | "fatal" | "off";
```

Comment:
Thoughts on providing some helper levels?
```suggestion
export type LogLevel = "all" | "trace" | "debug" | "detail" | "info" | WarnLogLevel | ErrorLogLevel | "off";
export type ErrorLogLevel = "error" | "fatal";
export type WarnLogLevel = "warn";
```

Reply:
This makes sense, but I think it will make the code much more difficult to read. Also even with this we can't be sure we're covering all the error levels (or warn levels) with the arrays.